core/pkg/utils/files.go

```diff
@@ -18,6 +20,31 @@ func FileExists(path string) (bool, error) {
 	return true, nil
 }
 
+func CopyFile(src, dst string) error {
+	source, err := os.Open(src)
+	if err != nil {
+		return fmt.Errorf("failed to open source file: %w", err)
+	}
+	defer source.Close()
+
+	destination, err := os.Create(dst)
+	if err != nil {
+		return fmt.Errorf("failed to create destination file: %w", err)
+	}
+	defer destination.Close()
+
+	_, err = io.Copy(destination, source)
+	if err != nil {
+		return err
+	}
+
+	if err := destination.Sync(); err != nil {
```

Comment:
is this necessary if this function calls `Close()`?

Reply:
I must have had some reason to use it in my original function but I'm suspicious of putting this in a common function where it could potentially have a significant performance penalty. I'll remove it.